pyqir-generator/pyqir/generator/_values.py

```diff
@@ -1,11 +1,11 @@
-# Copyright (c) Microsoft Corporation.
-# Licensed under the MIT License.
-
-from pyqir.generator._native import Qubit, ResultRef
-from typing import Union
-
-Value = Union[bool, int, float, Qubit, ResultRef]
-"""
-A QIR or LLVM value, or a Python value that can be automatically converted into
-one.
-"""
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from pyqir.generator._native import Qubit, ResultRef, Value
+from typing import Union
+
+Valuable = Union[Value, bool, int, float, Qubit, ResultRef]

```

Comment:
The Value type is the true representation of a value, and it's a Python wrapper around a Rust enum for each type of value (including variables). The rest of the Python types we support automatically converting into a Value for convenience. "Valuable" is just my cute name for "able to be (converted into) a Value", but let me know if you have a better name.